fastlane/Fastfile

```diff
@@ -44,26 +57,111 @@ end
 default_platform :ios
 
 platform :ios do
+  # Builds the app binary for Enterprise distribution (Prototype Build)
+  #
+  # @param fetch_code_signing [Boolean] If true, will call `configure_code_signing_enterprise` before building. Recommended on CI.
+  # @param build_number [String] The build number to use for the build. When in CI, defaults to the value of `BUILDKITE_BUILD_NUMBER` environment variable.
+  # @param version [String] The version to use for the build. When in CI, defaults to the PR number, falling back to the branch name.
+  #
+  lane :build_prototype do |fetch_code_signing: true, build_number: build_number_from_ci, version: pr_number_or_branch_name_from_ci|
```

Comment:
Maybe it's just me, but I like how this implementation does not have Buildkite hardcoded into it. By requiring `build_number` and `version` as call parameters, and by feeding them via dedicated methods (`build_number_from_ci` and `pr_number_or_branch_name_from_ci`) I feel the code is more separated, in particular the difference between CI the concept and Buildkite the provider.